### ISSUE TITLE
Improve tests for record deletion

### ISSUE THREAD

**Issue by peterkelly** [hyperdns/hapi-vendor-toolkit#1]
There's currently a bug in the Google driver whereby if you delete a record, it actually deletes all the other records in the same record set (that is, for all other records with the same name and type). This is not currently picked up by the tests, but we should have a test case for this.

**Comment by ewelton** [comment]
Yeah - agreed.  Please update the reference test script - we can break that down into methods if we want, but I don't like the idea of following the "UnitTest" model.  The UnitTest model (which I tried) leads to excessive tear-up, tear-down repetition, and results in huge overhead.  It seemed like having a "single, well orchestrated, uber-test" worked out best.

As such, there is the question of "decomposing the test into parts, but parts that are executed linearly and with expectation of completion of the previous parts".

definitely - we should have caught that.! ;)
